third_party/machinarium/sources/clock.c

```diff
@@ -39,45 +40,81 @@ void mm_clock_free(mm_clock_t *clock)
 	mm_buf_free(&clock->timers);
 }
 
+// Binary search routine for timer in sorted array.
+// We expect that there is at most only one occurrence of timer in list.
+// If there is not timer in list, we must return first index greater than timer.
+static int mm_clock_get_insert_position(mm_timer_t **list, int count, mm_timer_t *timer)
+{
+    int low = 0, high = count;
+    while (low < high)
+    {
+        int med = low + (high - low) / 2;
+        int cmp = mm_clock_cmp(timer, list[med]);
+        if (cmp == 0)
+	        return med;
+        if (cmp > 0) {
+            low = med + 1;
```

Comment:
what about 1 item list with list[med] > timer ? 
med + 1 whould be 1, that's out of list, should be 0 ?